extensions/wikia/AdEngine/js/run/desktop.run.js

```diff
@@ -89,7 +91,20 @@ require([
 				adEngine.run(adConfigDesktop, win.ads.runtime.sp.slots, 'queue.sp');
 			});
 		}
+
 		sourcePoint.initDetection();
+
+		if (context.opts.recoveredAdsMessage) {
+			loader({
+				type: loader.AM_GROUPS,
+				resources: ['adengine2_ads_recovery_message_js']
+			}).done(function () {
+				require(['ext.wikia.adEngine.recovery.message'], function (recoveredAdMessage) {
+					recoveredAdMessage.init();
+				});
+			});
```

Comment:
I don't think you're wrong. However, I'd go with this solution until we launch it globally then it'd make sense to load it for every page view, I guess.